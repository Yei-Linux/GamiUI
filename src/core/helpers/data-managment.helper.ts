export const mapEnum = (enumerable: any): any[] => {
  const enumMembers: any[] = Object.keys(enumerable).map(
    (key) => enumerable[key]
  )
  return enumMembers
}
