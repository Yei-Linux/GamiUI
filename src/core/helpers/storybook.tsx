import React from "react";
import TableStories from "../../styles/utilities/TableStories";
import { options as optionsStyle } from "../utils/constants";

export const getTemplate = (Component: React.ComponentType<any>): any => (
  args: any
) => {
  return <Component {...args} />;
};

export const getListTemplate = (Component: React.ComponentType<any>): any => ({
  items,
  field,
  ...args
}: {
  items: any[];
  field: any;
}) =>
  items.map((item, index) => (
    <TableStories key={index} item={item} field={field}>
      <Component {...args} {...item} />
    </TableStories>
  ));

export const stylesControl = {
  style: { control: "object" },
  textAlign: { control: { type: "select", options: optionsStyle.textAlign } },
  fontWeight: {
    control: { type: "select", options: optionsStyle.fontWeight },
  },
  width: { control: { type: "select", options: optionsStyle.width } },
  shadow: { control: { type: "select", options: optionsStyle.shadow } },
  border: { control: { type: "select", options: optionsStyle.border } },
  heigth: { control: { type: "select", options: optionsStyle.height } },
};