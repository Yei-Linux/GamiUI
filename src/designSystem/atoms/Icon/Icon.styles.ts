import styled from "@emotion/styled";
import { twinStyles } from "../../../styles/utilities/twinStyles";

export const IconWrapper = styled.i<{
  border?: string;
  shadow?: string;
}>`
  border: none;
  outline: none;
  cursor: pointer;
  appearance: none;
  display: flex;
  justify-content: center;
  align-items: center;

  ${(props: any) => twinStyles(props)};
`;
