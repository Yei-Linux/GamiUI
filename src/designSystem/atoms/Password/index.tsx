export { default } from "./Password";
export { options } from "./constants";
