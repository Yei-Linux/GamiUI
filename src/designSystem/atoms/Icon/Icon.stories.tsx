import { Meta } from "@storybook/react";
import { getTemplate, stylesControl } from "../../../core/helpers/storybook";

import Icon from ".";
import { options as optionsIcon } from "./constants";

const Template = getTemplate(Icon);

export default {
  title: "Atoms/Icon",
  component: Icon,
  argTypes: {
    children: { control: "text" },
    name: { control: { type: "select", options: optionsIcon.names } },
    ...stylesControl,
  },
} as Meta;

export const BasicIcon = Template.bind({});
BasicIcon.args = {
  fill: "red",
  size: "25px",
  name: "facebook",
};
