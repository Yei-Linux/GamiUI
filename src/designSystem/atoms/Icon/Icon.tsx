import React, { useEffect, useState } from "react";
import { IGeneralProps } from "../../../core/domain/interfaces/IGeneralProps";
import { IconNames } from "../../../core/domain/types";
import { IconsPack } from "./constants";
import { IconWrapper } from "./Icon.styles";

interface IIcon extends IGeneralProps {
  /**
   * Fill Icon
   */
  fill: string;
  /**
   * Name
   */
  name: IconNames;
  /**
   * Size icon
   */
  size: string;
}

const Svg = ({
  children,
  ...props
}: {
  children: React.ReactNode;
  width: string;
  height: string;
  viewBox: string;
  fill: string;
}) => {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" {...props}>
      {children}
    </svg>
  );
};

const Icon = ({ fill, name, size, ...args }: IIcon) => {
  const [icon, setIcon] = useState(IconsPack[name]);

  useEffect(() => {
    setIcon(IconsPack[name]);
  }, [name]);

  return (
    <IconWrapper {...args}>
      <Svg fill={fill} width={size} height={size} viewBox={icon.viewBox}>
        {icon.svg}
      </Svg>
    </IconWrapper>
  );
};

Icon.defaultProps = {
  fill: "none",
  name: "facebook",
  size: "16px",
};

export default Icon;