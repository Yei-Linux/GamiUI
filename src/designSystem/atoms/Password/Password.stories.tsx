import React from "react";
import { Meta } from "@storybook/react";
import { getTemplate, stylesControl } from "../../../core/helpers/storybook";

import Password from ".";
import Icon from "../Icon";

const Template = getTemplate(Password);

export default {
  title: "Atoms/Password",
  component: Password,
  args: { ...Password.defaultProps },
  argTypes: {
    positionPrefix: {
      control: "select",
      options: ["left", "right"],
    },
    placeholder: { control: "text" },
    prefix: {
      options: ["a", "b"],
      mapping: {
        a: null,
        b: <Icon fill="#374a54" />,
      },
      control: {
        type: "select",
        labels: {
          a: "withoutIcon",
          b: "withIcon",
        },
      },
    },
    ...stylesControl,
  },
} as Meta;

export const BasicPassword = Template.bind({});
BasicPassword.args = {
  placeholder: "Type your password",
  width: "NORMAL",
  heigth: "SMALL",
  shadow: "",
  border: "MEDIUM"
};