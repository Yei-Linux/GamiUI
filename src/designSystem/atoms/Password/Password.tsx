import React, { useState } from "react";
import Icon from "../Icon";
import Input, { IInput } from "../Input/Input";

export interface IPassword extends IInput {
  iconVisible: React.ReactNode;
  iconHide: React.ReactNode;
}
const Password = ({ iconHide, iconVisible, ...args }: IPassword) => {
  const [isVisible, setIsVisible] = useState(false);

  const handleToggle = () => setIsVisible(!isVisible);

  const toggleType = () => (isVisible ? "input" : "password");

  const renderIconPass = () => (
    <div onClick={() => handleToggle()}>
      {isVisible ? iconVisible : iconHide}
    </div>
  );

  return <Input type={toggleType()} prefix={renderIconPass()} {...args} />;
};

export default Password;

Password.defaultProps = {
  positionPrefix: "right",
  iconVisible: <Icon fill="red" size="30px" name="openeye" />,
  iconHide: <Icon fill="blue" size="30px" name="closeeye" />,
  width: "NORMAL",
  heigth: "SMALL",
  border: "MEDIUM",
};
