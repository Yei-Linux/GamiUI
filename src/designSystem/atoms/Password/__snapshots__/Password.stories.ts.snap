// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/Password Basic Password 1`] = `
<div
  autoComplete="off"
  className="css-11c3v5m-InputWrapper e7vlt9n0"
  name="gamification"
  placeholder="Type your password"
  prefix={
    <div
      onClick={[Function]}
    >
      <Icon
        fill="blue"
        name="closeeye"
        size="30px"
      />
    </div>
  }
  readOnly={false}
  type="password"
  width="NORMAL"
>
  <div
    onClick={[Function]}
  >
    <i
      className="css-6tw2qr-IconWrapper e1cr9qb30"
    >
      <svg
        fill="blue"
        height="30px"
        viewBox="0 0 512.52 512.52"
        width="30px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <g>
          <g>
            <path
              d="m512.52 187.167-3.34 6.69c-.71 1.45-18.07 35.74-57.99 69.95-23.36 20.02-49.72 35.97-78.35 47.4-35.65 14.24-74.87 21.46-116.58 21.46s-80.93-7.22-116.58-21.46c-28.63-11.43-54.99-27.38-78.35-47.4-39.92-34.21-57.28-68.5-57.99-69.95l-3.34-6.69 3.34-6.69c.71-1.44 18.07-35.74 57.99-69.95 23.36-20.01 49.72-35.96 78.35-47.4 35.65-14.24 74.87-21.46 116.58-21.46s80.93 7.22 116.58 21.46c28.63 11.44 54.99 27.39 78.35 47.4 39.92 34.21 57.28 68.51 57.99 69.95z"
              fill="#fbf3f3"
            />
            <path
              d="m512.52 187.167-3.34 6.69c-.71 1.45-18.07 35.74-57.99 69.95-23.36 20.02-49.72 35.97-78.35 47.4-35.65 14.24-74.87 21.46-116.58 21.46v-291c41.71 0 80.93 7.22 116.58 21.46 28.63 11.44 54.99 27.39 78.35 47.4 39.92 34.21 57.28 68.51 57.99 69.95z"
              fill="#e8e3f0"
            />
            <ellipse
              cx="256.26"
              cy="187.167"
              fill="#adcfff"
              rx="114.12"
              ry="114.12"
              transform="matrix(.707 -.707 .707 .707 -57.291 236.023)"
            />
            <path
              d="m370.38 187.167c0 63.03-51.09 114.12-114.12 114.12v-228.24c63.03 0 114.12 51.101 114.12 114.12z"
              fill="#6aa8ff"
            />
            <ellipse
              cx="256.26"
              cy="189.437"
              fill="#363671"
              rx="59.42"
              ry="59.42"
              transform="matrix(.257 -.967 .967 .257 7.395 388.49)"
            />
            <path
              d="m315.68 189.437c0 32.82-26.6 59.42-59.42 59.42v-118.84c32.82 0 59.42 26.6 59.42 59.42z"
              fill="#16164d"
            />
            <circle
              cx="298.311"
              cy="151.391"
              fill="#fbf3f3"
              r="28.877"
            />
          </g>
          <ellipse
            cx="412.11"
            cy="370.703"
            fill="#ff7c48"
            rx="100.15"
            ry="100.15"
            transform="matrix(.707 -.707 .707 .707 -141.422 399.982)"
          />
          <path
            d="m512.26 370.703c0 55.22-44.93 100.15-100.15 100.15v-200.3c55.22 0 100.15 44.92 100.15 100.15z"
            fill="#ff3939"
          />
          <path
            d="m412.11 355.707 10 14.996-10 15.004h-41.48v-30z"
            fill="#fbf3f3"
          />
          <path
            d="m412.11 355.707h41.48v30h-41.48z"
            fill="#e8e3f0"
          />
        </g>
      </svg>
    </i>
  </div>
  <input
    autoComplete="off"
    className="css-7dwcir-InputContainer e7vlt9n1"
    name="gamification"
    onChange={[Function]}
    placeholder="Type your password"
    prefix={
      <div
        onClick={[Function]}
      >
        <Icon
          fill="blue"
          name="closeeye"
          size="30px"
        />
      </div>
    }
    readOnly={false}
    type="password"
    width="NORMAL"
  />
</div>
`;
